src/Simplify_Call.cpp

```diff
@@ -50,15 +64,18 @@ Expr Simplify::visit(const Call *op, ExprInfo *bounds) {
                         return mutate(Div::make(a, b), bounds);
                     }
                 } else {
+                    // We know that all the meaningful bits will be shifted away;
+                    // just replace with a constant 0 or -1 if possible.
                     if (shift_left) {
                         return make_zero(t);
                     } else {
                         // shift-right-out-of-bounds -> zero for uint.
                         if (t.is_uint()) {
                             return make_zero(t);
+                        } else {
+                            // shift-right-out-of-bounds -> zero or -1 for int.
+                            return mutate(select(a < 0, make_const(t, -1), make_zero(t)), bounds);
```

Comment:
It seems like leaving this as a shift is better. It's the same number of the same kind of instructions, but doesn't need the constant registers.

Reply:
The motivation for this construct is to constant-fold expressions that can result from the recent changes to Bounds (e.g. 32767 << -32768); if we left it as a shift, we'd end up with `shift_right(32767, 32768)` in our Expr tree post-simplify.

Reply:
So we know a is a constant? If so, how about a compile time if, instead of a runtime if?

Reply:
Although I see now this code is pre-existing, so maybe my comment isn't relevant in this PR.

Reply:
That code is pre-existing by only a few days. 

I suppose we could structure it to only do that if `a` is a constant, or to do something like

   if (can_prove(a < 0)) // -1
   else if (can_prove(a >= 0)) // 0
   else // use shift

Reply:
(oh, and `a` is **not** known to be a constant.)